### ISSUE TITLE
About convergence on the celebahq dataset

### ISSUE THREAD

**Issue by Ellohiye** [SHI-Labs/FcF-Inpainting#6]
Hello author! Thanks for your wonderful work! I would like to ask how many epochs of training the model on celebahq can converge when the equipment is limited? I only have one gpu, so the training time is very important for me.
Looking forward to your reply！

**Comment by Ellohiye** [comment]
One more question I have is can FFC be used for image training with small masks？Looking forward to your reply！

**Comment by praeclarumjj3** [comment]
Hi @yunbingbing, thanks for your interest in our work. To answer your questions:
- `Re: single-gpu`: 
     - We train celeba-hq for 25M images (set `--kimg` as 25000 in your training command) on 8 GPUs. 
     - Because you are using a single GPU, I assume you would have to decrease the batch size, which can affect the convergence. I suggest you train for `10000 kimg` and closely look at the FID score plot during training. You might need to tune some hyper-parameters (lr, etc.) for optimal performance.
- `Re: small masks`: I assume you mean lower-resolution input images? If that's the case, you can train with "small masks."  You may need to optimize the number of FFC blocks at each resolution stage though: https://github.com/SHI-Labs/FcF-Inpainting/blob/afad2bc2ec62f38b5478a124f3550621345612b9/training/models.py#L477

**Comment by Ellohiye** [comment]
> Hi @yunbingbing, thanks for your interest in our work. To answer your questions:
> 
> * `Re: single-gpu`:
>   
>   * We train celeba-hq for 25M images (set `--kimg` as 25000 in your training command) on 8 GPUs.
>   * Because you are using a single GPU, I assume you would have to decrease the batch size, which can affect the convergence. I suggest you train for `10000 kimg` and closely look at the FID score plot during training. You might need to tune some hyper-parameters (lr, etc.) for optimal performance.
> * `Re: small masks`: I assume you mean lower-resolution input images? If that's the case, you can train with "small masks."  You may need to optimize the number of FFC blocks at each resolution stage though: https://github.com/SHI-Labs/FcF-Inpainting/blob/afad2bc2ec62f38b5478a124f3550621345612b9/training/models.py#L477

Thank you very much for your reply! I have two more questions:
(1) Regarding the small mask, I actually mean that the model can be trained with images covered by the small area mask? Is FFC only suitable for image training with large area mask?
(2) Regarding the celebahq dataset, I still have a question. The training uses a resolution of 256*256 for training. Can we use  512*512 size images for testing? Which hyperparameters of the code should I modify if high-resolution testing is possible?
Looking forward to your reply! Thanks!

**Comment by praeclarumjj3** [comment]
Hi @yunbingbing,

- `Re: small masks`: Most definitely. FFC illustrates its benefits more firmly with larger masks. At the same time, it is possible and suitable to train with small masks.
- `Re: 512x512`: We build our method on top of the StyleGAN2 generator, which does not support inputs with resolutions different than that used during training. If you want to test on higher-resolution images, you would need to train a new model.

**Comment by Ellohiye** [comment]
> Hi @yunbingbing,
> 
> * `Re: small masks`: Most definitely. FFC illustrates its benefits more firmly with larger masks. At the same time, it is possible and suitable to train with small masks.
> * `Re: 512x512`: We build our method on top of the StyleGAN2 generator, which does not support inputs with resolutions different than that used during training. If you want to test on higher-resolution images, you would need to train a new model.

thank you for your reply! This is very helpful to me!
